### PR TITLE
Updates Nextjs to version 9

### PR DESCRIPTION
**Pull request by antonjb**
Updates to the latest TypeScript version. Updates Next from v8 to v9.
Followed Next's upgrade guide. Most changes are fixing Type errors
resulting from the upgrade. Added TSC build task to quickly check TS
errors across the project.

### CODE REVIEW

**Comment by mzaatar** [comment]
Looks like the build is failing.
`Build failed because of webpack errors
2020-02-15T11:30:44.9185938Z     at build (d:\a\1\s\node_modules\next\dist\build\index.js:10:900)
2020-02-15T11:30:44.9186198Z npm ERR! code ELIFECYCLE
2020-02-15T11:30:44.9186540Z npm ERR! errno 1
2020-02-15T11:30:44.9186900Z npm ERR! dddperth-website@1.0.0 build: `next build`
2020-02-15T11:30:44.9187204Z npm ERR! Exit status 1
2020-02-15T11:30:44.9187453Z npm ERR! 
2020-02-15T11:30:44.9187679Z npm ERR! Failed at the dddperth-website@1.0.0 build script.
2020-02-15T11:30:44.9187963Z npm ERR! This is probably not a problem with npm. There is likely additional logging output above.`

File: .vscode/tasks.json [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
+      "group": {
+        "kind": "build",
+        "isDefault": true
+      },
+      "command": "yarn",
+      "problemMatcher": "$tsc",
+      "args": ["tsc"]
+    },
+    {
+      "type": "npm",
```
Comment (robdmoore) [inline .vscode/tasks.json]: Might as well remove this?

Comment (antonjb) [reply .vscode/tasks.json]: Yeah, we could actually.

File: pages/_app.tsx [hunk 2]
```diff
@@ -1,30 +1,13 @@
...
-import App, { Container } from 'next/app'
 import React from 'react'
 import { theme } from '../components/utils/styles/theme'
 
-class CustomApp extends App {
-  static async getInitialProps({ Component, ctx }) {
-    let pageProps = {}
-
-    if (Component.getInitialProps) {
-      pageProps = await Component.getInitialProps(ctx)
```
Comment (robdmoore) [inline pages/_app.tsx]: I assume this is now just part of the default next App base class?

Comment (antonjb) [reply pages/_app.tsx]: They removed it. You only need that with blocking script loading. All we were doing is adding the theme provider but you had to include that whereas now you don't.

**Comment by robdmoore** [comment]
You may want to double check that. This is code I added to ensure that server side rendering worked for all components I.e. the getInitialProps method was always called.

You can check by going directly (i.e. not via client side navigation) to pages that take props from server side render (like agenda?) directly and make sure they still work.

--



On 16 Feb 2020, at 1:41 pm, Anton <notifications@github.com> wrote:

﻿

@antonjb commented on this pull request.

________________________________

In pages/_app.tsx<https://github.com/dddwa/dddperth-website/pull/248#discussion_r379876857>:

>  import React from 'react'
 import { theme } from '../components/utils/styles/theme'

-class CustomApp extends App {
-  static async getInitialProps({ Component, ctx }) {
-    let pageProps = {}
-
-    if (Component.getInitialProps) {
-      pageProps = await Component.getInitialProps(ctx)


They removed it. You only need that with blocking script loading. All we were doing is adding the theme provider but you had to include that whereas now you don't.

**Comment by antonjb** [comment]
@robdmoore Next's guide says the setup we had could be removed. I'll double check the SSR though.
https://github.com/zeit/next.js/blob/canary/UPGRADING.md#check-your-custom--pages_appjs

**Comment by robdmoore** [comment]
Brilliant!

--



On 16 Feb 2020, at 6:41 pm, Anton <notifications@github.com> wrote:

﻿

@robdmoore<https://github.com/robdmoore> Next's guide says the setup we had could be removed. I'll double check the SSR though.
https://github.com/zeit/next.js/blob/canary/UPGRADING.md#check-your-custom--pages_appjs

**Pull request merged**